### ISSUE TITLE
Stream data from Hugging Face

### ISSUE THREAD

**Issue by lhoestq** [webdataset/webdataset#265]
Hi ! I'm Quentin from HF

We're fans and users of webdataset and I think streaming from `hf://` URLs would be a nice addition ! There are 30k public datasets shared by the community on HF, including many webdatasets :)

Implementation wise: from the different implementations of `gopen` it looks like the recommended way is to use a CLI and `cat` the data, is that correct ?

**Comment by tmbdev** [comment]
Yes, that would be great! I love Huggingface and use it a lot, also in teaching.

The recommended way is indeed a command line program for fetching the data. This gives us asynchronous I/O for free and tends to result in fewer dependencies and to be easier to debug. (However, if you have a library, you can use that as well.)

I assume that caching is probably useful for a lot of your use cases. You could implement that at the level of the command line program, or you could take advantage of the built-in cache mechanism.

Please let me know if you need help and/or just submit a PR.

**Comment by lhoestq** [comment]
Btw we just added a WebDataset Viewer on HF !

See https://huggingface.co/datasets/timm/imagenet-12k-wds

Would love to have some feedback @tmbdev :)

<img width="1041" alt="image" src="https://github.com/webdataset/webdataset/assets/42851186/5da03b24-645a-4f07-83a8-63d2c03b9bfb">

<img width="1032" alt="image" src="https://github.com/webdataset/webdataset/assets/42851186/b56b713b-2e39-4854-8af5-13b3f871d9a5">
